src/hooks/fix-decompress.ts

```diff
@@ -0,0 +1,93 @@
+import zlib from 'zlib';
+import { ClientRequest, IncomingMessage } from 'http';
+import { HandlerFunction } from 'got-cjs';
+import { PassThrough, Transform } from 'stream';
+import mimicResponse from 'mimic-response';
+
+const onResponse = (response: IncomingMessage, propagate: (fixedResponse: IncomingMessage) => void) => {
+    const encoding = response.headers['content-encoding']?.toLowerCase();
+
+    // Append empty chunk.
+    const zlibOptions = {
+        flush: zlib.constants.Z_SYNC_FLUSH,
+        finishFlush: zlib.constants.Z_SYNC_FLUSH,
+    };
+
+    const useDecompressor = (decompressor: Transform) => {
+        delete response.headers['content-encoding'];
+
+        const result = new PassThrough({
+            autoDestroy: false,
+            destroy(error, callback) {
+                response.destroy();
+
+                callback(error);
+            },
+        });
+
+        response.pipe(decompressor).pipe(result);
```

Comment:
Are errors properly handled? I don't see any handlers anywhere.

Reply:
> Please note that myStream and responseStream never throw. The error is passed to the request instead.

Is this the answer? (from mimic docs)

Reply:
Ah, missed [this](https://github.com/sindresorhus/decompress-response/blob/c717c55f958ab492d13338cc999eb7268d82e1fb/index.js#L45) line. The empty check is unnecessary because we're incorrectly (but on purpose) using Z_SYNC_FLUSH.